python/kfserving/kfserving/kfmodel_repository.py

```diff
@@ -31,19 +32,28 @@ def __init__(self, models_dir: str = MODEL_MOUNT_DIRS):
     def set_models_dir(self, models_dir):  # used for unit tests
         self.models_dir = models_dir
 
-    def get_model(self, name: str) -> Optional[KFModel]:
+    def get_model(self, name: str) -> Optional[Union[KFModel, RayServeHandle]]:
         return self.models.get(name, None)
 
-    def get_models(self) -> List[KFModel]:
+    def get_models(self) -> List[Union[KFModel, RayServeHandle]]:
         return list(self.models.values())
 
     def is_model_ready(self, name: str):
         model = self.get_model(name)
-        return False if model is None else model.ready
+        if not model:
+            return False
+        if isinstance(model, KFModel):
+            return False if model is None else model.ready
+        else:
+            # TODO find out how to get the remote model status
```

Comment:
For Ray Serve, the models are guaranteed to be ready after the deploy call. So `ModelClass.deploy()` itself is blocking and it will wait for `num_replicas` models to be ready. You can use `_wait_for_goal` API to make it async:
```
from serve.api import _get_global_client

client = serve.start()
goal_id = Model.deploy(_blocking=False)
is_ready: bool = client._wait_for_goal(goal_id=goal_id, timeout=1) # This timeout is configurable, and it will return True or False indicating whether or not the goal has been completed. It can even be 0.
```